src/compiler/checker.ts

```diff
@@ -1412,6 +1411,14 @@ namespace ts {
                         return lastLocation.symbol;
                     }
                 }
+                if (originalLocation && isInJavaScriptFile(originalLocation) && originalLocation.parent) {
+                    if (isIdentifier(originalLocation) && isPropertyAccessExpression(originalLocation.parent) &&
+                        originalLocation.escapedText === "module" && originalLocation.parent.name.escapedText === "exports") {
+                        const moduleSymbol = createSymbol(SymbolFlags.FunctionScopedVariable, "module" as __String, CheckFlags.ModuleExports);

```

Comment:
I think the creation should be cached somewhere, since we base several other caches on the symbol. If we create 100 different symbols, we'll also create a new type for each of those symbols.

Reply:
Yikes! Good catch.

Reply:
This is equivalent to always binding `module` in `module.exports` to a specific symbol, so I just moved it to the binder and moved `ModuleExports` from CheckFlags to SymbolFlags.